client/app/lib/helpers/download.js

```diff
@@ -0,0 +1,55 @@
+const DELAY = 500; // Interval to check for job completion in ms.
```

Comment:
This can be reduced to 2000ms to reduce the requests to server? Because it's not performance critical and normally download will take quite some time.